client/dive-common/components/Viewer.vue

```diff
@@ -355,16 +335,26 @@ export default defineComponent({
         datasetType.value = meta.type as DatasetType;
 
         const trackData = await loadDetections(datasetId.value, props.revision);
-        const tracks = Object.values(trackData);
-        progress.total = tracks.length;
+        const tracks = Object.values(trackData.tracks);
+        const groups = Object.values(trackData.groups);
+        progress.total = tracks.length + groups.length;
         for (let i = 0; i < tracks.length; i += 1) {
           if (i % 4000 === 0) {
           /* Every N tracks, yeild some cycles for other scheduled tasks */
             progress.progress = i;
             // eslint-disable-next-line no-await-in-loop
             await new Promise((resolve) => window.setTimeout(resolve, 500));
           }
-          insertTrack(Track.fromJSON(tracks[i]), { imported: true });
+          trackStore.insert(Track.fromJSON(tracks[i]), { imported: true });
+        }
+        for (let i = 0; i < groups.length; i += 1) {
+          if (i % 4000 === 0) {
+          /* Every N tracks, yeild some cycles for other scheduled tasks */
+            progress.progress = tracks.length + i;
+            // eslint-disable-next-line no-await-in-loop
+            await new Promise((resolve) => window.setTimeout(resolve, 500));
+          }
+          groupStore.insert(Group.fromJSON(groups[i]));
```

Comment:
Needs the imported argument added to it to prevent upserting all the groups each save.